### PR TITLE
Refactor/maps-login-main-market

### PR DESCRIPTION
**Pull request by Zeavee**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/login/LoginTest.java [hunk 1]
```diff
@@ -61,13 +60,13 @@ public void writingEmail_Works() {
...
-/*
+
     @Test
     public void login_shouldWorkWithRegisteredUser() {
         MissingFieldTestFactory.testFieldTwoActionsCloseKeyboard(typeText(email), typeText(password), R.id.emaillog, R.id.passwordlog);
         onView(withId(R.id.loginButton)).perform(click());
-        onView(withId(R.id.logoutBt)).perform(click());
-    }*/
+        onView(withId(R.id.solo)).check(matches(isDisplayed()));
+    }
```
Comment (kangpl) [inline app/src/androidTest/java/ch/epfl/sdp/login/LoginTest.java:69]: Good to bring the test back

File: app/src/main/java/ch/epfl/sdp/game/Game.java [hunk 2]
```diff
@@ -13,6 +13,14 @@
...
  */
 public class Game implements Updatable {
+    /**
+     * This boolean tells if the game started and is used in the GoogleLocationFinder to start the game only one time
+     */
+    public boolean gameStarted = false;
+    /**
+     * This is the instance of the start game controller of the game
+     */
+    public StartGameController startGameController;
```
Comment (kangpl) [inline app/src/main/java/ch/epfl/sdp/game/Game.java:23]: Thanks for this nice debugging

File: app/src/androidTest/java/ch/epfl/sdp/SocialTests/AddFriendsAndChatActivityTest.java [hunk 3]
```diff
@@ -96,21 +113,79 @@ public void step3() {
...
+        onView(withId(R.id.addFriendsBackButton)).perform(click());
+        Thread.sleep(2000);
+        chatStep1();
+        chatStep2();
+        chatStep3();
+        chatStep4();
+    }
+
+    // click on the chat button
+    public void chatStep1() {
```
Comment (varathar53) [inline app/src/androidTest/java/ch/epfl/sdp/SocialTests/AddFriendsAndChatActivityTest.java:140]: it looks complicated. Does it have to be like this? Its recursive?

File: app/src/main/java/ch/epfl/sdp/login/LoginFormActivity.java [hunk 4]
```diff
@@ -31,20 +34,23 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // If the user has already logged in, go to MainActivity directly
         if (authenticationAPI.getCurrentUserEmail() != null) {
-            startActivity(new Intent(LoginFormActivity.this, MainActivity.class));
+            startActivity(new Intent(LoginFormActivity.this, MainMenuActivity.class));
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/login/LoginFormActivity.java:37]: better use findbyId

File: app/src/main/java/ch/epfl/sdp/market/MarketActivity.java [hunk 5]
```diff
@@ -124,7 +124,8 @@ private void syncMarketBackendToFrontend() {
         if (backend == null) finish();
         setupItemToViewMap();
         for (Item i : backend.getStock().keySet()) {
-            ((TextView) findViewById(itemToViewMap.get(i.getClass().hashCode()))).setText("Cost: " + backend.getStock().get(i).second + "\n" + "Value: " + (((int) (100 * i.getValue())) / 100.0));
+            String newText = "Cost: " + backend.getStock().get(i).second + "\n" + "Value: " + (((int) (100 * i.getValue())) / 100.0);
```
Comment (varathar53) [inline app/src/main/java/ch/epfl/sdp/market/MarketActivity.java:127]: is it not better to shorten this string?

File: app/src/test/java/ch/epfl/sdp/game/GameTest.java [hunk 6]
```diff
@@ -122,11 +122,8 @@ public void gameThread_runs() {
     public void gameThreadCatchExceptions() {
         Game.getInstance().initGame();
 
-        Updatable updatable = new Updatable() {
-            @Override
-            public void update() {
-                double a = 1 / 0;
-            }
+        Updatable updatable = () -> {
+            double a = 1 / 0;
```
Comment (varathar53) [inline app/src/test/java/ch/epfl/sdp/game/GameTest.java:126]: why 1/0? Its undefined?

File: app/src/main/java/ch/epfl/sdp/login/LoginFormActivity.java (hunk 4 above)
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/login/LoginFormActivity.java:37]: I use findById, and this is your doings ^^

File: app/src/main/java/ch/epfl/sdp/market/MarketActivity.java (hunk 5 above)
Comment (Zeavee) [reply app/src/main/java/ch/epfl/sdp/market/MarketActivity.java:127]: Same comment (this branch was merged into the other one)

File: app/src/test/java/ch/epfl/sdp/game/GameTest.java (hunk 6 above)
Comment (Zeavee) [reply app/src/test/java/ch/epfl/sdp/game/GameTest.java:126]: It's to give an exception for test coverage

**Pull request merged**